planpro_importer/planpro110/nodereader.py

```diff
@@ -0,0 +1,117 @@
+import logging
+
+from yaramo.model import DbrefGeoNode, Node, Topology
+
+from .model110 import CContainer
+
+
+class NodeReader:
+
+    def __init__(self, topology: Topology, container: CContainer):
+        """The node reader reads all nodes from a container and adds them
+        to the topology.
+
+        :param topology: The topology
+        :param container: The container
+        """
+        self.topology: Topology = topology
+        self.container: CContainer = container
+
+    def read_nodes(self):
+        """Read the nodes from the container."""
+
+        for top_knoten in self.container.TOP_Knoten:
+            node_obj = Node(uuid=top_knoten.Identitaet.Wert)
+
+            # Coordinates
+            geo_node_uuid = top_knoten.ID_GEO_Knoten.Wert
+            x, y = self.get_coordinates_of_geo_node(self.container, geo_node_uuid)
+            if x is None or y is None:
+                continue
+            node_obj.geo_node = DbrefGeoNode(x, y, uuid=geo_node_uuid)
+
+            self.topology.add_node(node_obj)
+
+    def add_point_names(self):
+        """Add the names of the points to the points. If there is no name defined,
+        it will use the last five characters of the UUID."""
+
+        for point_element in self.container.W_Kr_Gsp_Element:
+            element_uuid = point_element.Identitaet.Wert
+            point_name = point_element.Bezeichnung.Bezeichnung_Aussenanlage.Wert
+            component = self.get_component_by_element_uuid(element_uuid)
+            point = self.get_point_of_component(component)
+            if point is not None:
+                point.name = point_name
+
+        # Set default names for the rest
+        for node in self.topology.nodes.values():
+            if node.name is None:
+                node.name = node.uuid[-5:]
```

Comment:
Assuming a risk of name clashes here. Also assuming that it wouldn't be a big problem, except for humans trying to debug problems.

Reply:
Yes, shouldn't be a big problem.